File: 100-Days-Series/Day 69/MissingNumberInArray.c
#include <stdio.h>
// Function to check Missing number in an array
int getMissingNumber(int arr[], int size) {
    int n = size + 1; // Actual range (1 to n)
    int totalSum = n * (n + 1) / 2; // Sum of 1 to n
    int arraySum = 0;
    for (int i = 0; i < size; i++) {
        arraySum += arr[i];
    }
    return totalSum - arraySum;
}
// Let's Start Here
void main() {
    int arr[] = {1, 2, 4, 5, 6}; // Example
    int size = sizeof(arr) / sizeof(arr[0]); // Number of elements in the array
    int missingNumber = getMissingNumber(arr, size);
    printf("\nMissing number is: %d", missingNumber);
}

File: 100-Days-Series/Day 69/MissingNumberInArray.py

def get_missing_number(arr, size):
    n = size + 1  # Actual range (1 to n)
    total_sum = n * (n + 1) // 2  # Sum of 1 to n (integer division)
    array_sum = 0

    for num in arr:
        array_sum += num

    return total_sum - array_sum


# Let's Start Here
arr = [1, 2, 4, 5, 6]  # Example
size = len(arr)  # Number of elements in the array
missing_number = get_missing_number(arr, size)
print("Missing number is:", missing_number)
